power_port_vision.py

```diff
@@ -178,41 +176,64 @@ def set_camera_zoom_and_tilt(self):
         self.camera_manager.get_frame()
 
     def adjust_zoom_and_tilt(self):
-        if (
-            self.previous_target_min_x is not None
-            and self.previous_target_max_x is not None
-            and self.previous_target_top is not None
-        ):
+        if self.previous_power_port is not None:
             # First we compute the new zoom
             # A length in pixels l1 at zoom z1 is related to the length in pixels
             # l2 at zoom z2 by
             # l1/z1 = l2/z2, or l1/l2 = z1/z2, or z2 = z1*l2/l1
             # To compute a new zoom, we want to scale a length l at zoom z1 to a
             # max length l2 at the new zoom z2. Using the above:
             # new_zoom = old_zoom * l2 / l1
-            # The length we want to scale is the excursion from the centre of the
-            # image, and the length we want to scale to is MAX_EXCURSION, clipped
+            # The length we want to scale is either the horizontal excursion or
+            # the height of the target, depending on whether the width or height
+            # of the target is larger. The horizontal excurison is the distance
+            # from the minimum or maximum x from the centre of the image, and
+            # the length we want to scale to is the appropriate maximum, clipped
             # to the maximum and a minimum of 1.
-            min_from_centre = self.previous_target_min_x - FRAME_WIDTH / 2
-            max_from_centre = self.previous_target_max_x - FRAME_WIDTH / 2
-            excursion = max(abs(min_from_centre), abs(max_from_centre))
-            new_zoom = round(self.zoom_factor * self.MAX_EXCURSION / excursion, 2)
+            if (
+                self.previous_power_port.get_width()
+                > self.previous_power_port.get_height()
```

Comment:
I'm not sure about this, but should it check if the ratio is above/below 4:3?

Reply:
Hmmm. Very good observation. I'm not sure what the exact answer is, but this at least prevents the problems we were having with obliques. It's still an improvement, so I'm going to merge it, but I will add a comment with a TODO to come up with a better test.